### ISSUE TITLE
Update current architecture 

### ISSUE THREAD

**Issue by amosehiguese** [amosehiguese/subdomain-scanner#9]
Currently, the dnsresolve service is implemented using unary rpc. This causes a lot of latency while processing huge chunk of data since it's returning back the data as one huge chunk.

Take advantage of grpc stream feature to speed things up.

Also, you could use an executor to create a thread pool for handling tasks in an asynchronous way.

Note: based on the architecture on the readme, the dnsresolve doesn't yet interact with port scan service. That also needs to be fix.

Basically, that part of the application needs to be reimplemented based on the readme.

The frontend service should not interact with the port scan service like it does at the moment. It should get the final result from the dnsresolve service.
